modules/oracle/chain/bitcoin.go

```diff
@@ -148,6 +164,60 @@ func (b *bitcoinRelayer) ChainData(
 	return blocks, nil
 }
 
+// GetContractState implements chainRelay.
+func (b *bitcoinRelayer) GetContractState() (chainState, error) {
+	const queryKey = "blocklist"
+
+	var query struct {
+		GetStateByKeys json.RawMessage `graphql:"getStateByKeys(contractId: $contractId, keys: $keys)"`
+	}
+
+	variables := map[string]any{
+		"contractId": b.ContractID(),
+		"keys":       []string{queryKey},
+	}
+
+	var debugClient *http.Client
+	if os.Getenv("DEBUG") == "1" {
+		debugClient = &http.Client{
+			Transport: &loggingTransport{http.DefaultTransport},
+		}
+	}
+
+	client := graphql.NewClient(graphQLUrl, debugClient)
```

Comment:
This should be in the struct of the Bitcoin chain relay. It should not need to be initialized again every call.